docs/api-tutorials/custom-reporter.md

```diff
@@ -0,0 +1,113 @@
+
+The event names are exported from the `constants` property of `Mocha.Runner`:
+
+| Constant                 | Event Name  | Event Arguments | Description                                                                                 |
+| ------------------------ | ----------- | --------------- | ------------------------------------------------------------------------------------------- |
+| `RUNNER_EVENT_END`       | `end`       | _(n/a)_         | All [Suite]s, [Test]s and [Hook]s have completed execution                                  |
+| `RUNNER_EVENT_FAIL`      | `fail`      | `Test`, `Error` | A [Test] has failed or thrown an exception                                                  |
+| `RUNNER_EVENT_HOOK`      | `hook`      | `Hook`          | A [Hook] is about to execute                                                                |
+| `RUNNER_EVENT_HOOK_END`  | `hook end`  | `Hook`          | A [Hook] has completed execution                                                            |
+| `RUNNER_EVENT_PASS`      | `pass`      | `Test`          | A [Test] has passed                                                                         |
+| `RUNNER_EVENT_PENDING`   | `pending`   | `Test`          | A [Test] was skipped                                                                        |
+| `RUNNER_EVENT_RETRY`     | `retry`     | `Test`, `Error` | A [Test] failed, but is about to be retried; never emitted unless `retry` option is nonzero |
+| `RUNNER_EVENT_BEGIN`     | `start`     | _(n/a)_         | Execution will begin                                                                        |
+| `RUNNER_EVENT_SUITE`     | `suite`     | `Suite`         | The [Hook]s and [Test]s within a [Suite] are about to be executed                           |
+| `RUNNER_EVENT_SUITE_END` | `suite end` | `Suite`         | The [Hook]s and [Test]s within a [Suite] (and any children [Suite]s) completed execution    |
+| `RUNNER_EVENT_TEST`      | `test`      | `Test`          | A [Test] is about to be executed                                                            |
+| `RUNNER_EVENT_TEST_END`  | `test end`  | `Test`          | A [Test] has completed execution                                                            |
+| `RUNNER_EVENT_WAITING`   | `waiting`   | _(n/a)_         | Waiting for `global.run()` to be called; only emitted when `delay` option is `true`         |
+
+**Please use these constants** instead of the event names in your own reporter! This will ensure compatibility with future versions of Mocha.
+
+> It's important to understand that all suite callbacks will be run _before_ the [Runner] emits `RUNNER_EVENT_BEGIN`. Hooks and tests, however, won't run until _after_ the [Runner] emits `RUNNER_EVENT_BEGIN`.
+
+## Custom Reporter for Browser
+
+As of Mocha v6.0.0, custom reporters are _only_ "officially" supported in Node.js. [Mochify](https://npm.im/mochify) might help you. YMMV!
```

Comment:
Say what?

Reply:
To what are you referring?

Reply:
When did that happen?

Reply:
they've never been supported, as far as I'm aware.  might as well make that clear... I suppose I can remove the "as of.."

Reply:
What I was trying to say was this:

- Mocha supports 3rd-party reporters insofar as it provides an API for those reporters to use, and an option (`--reporter`) to run tests using one.
- The `Base` reporter--which Node.js reporters should extend--is not intended for use in a browser, so it would be difficult to make one portable.

But looking more closely at the code: while the HTML reporter is used by default in the browser, I'm not seeing anything that would prevent a user from using a custom reporter in a browser.  So I'll probably just remove this note and add another about `Base` & the browser.

- [ ] rewrite note about browser testing